### ISSUE TITLE
Update for newer versions of zabbix and papercut

### ISSUE THREAD

**Issue by TheFuzzyTech** [Open-Future-Belgium/zabbix-papercut#8]
This is sort of working with the latest version of Papercut and Zabbix. But the documentation isn't quite up to date. The Python script looks for the agent.exe which no longer exists when downloading the Zabbix source. You can get around this by downloading the agent.exe and adding it to the correct path in the archive, but not all data is collected properly in Zabbix. I'd be happy to help in any way to get this updated.

**Comment by Trikke76** [comment]
hi,
this would be helpful
i dont use papercut myself but plan to update it somewhere next year when i have my hands on a papercut server

**Comment by jonwitnoh** [comment]
Yes this is needed. I am unable to install it as a result

**Comment by Joffcom** [comment]
@jonwitnoh what happens when you try to install and use it? I happen to have a local PaperCut server running at the moment for another project.

**Comment by Kimotu** [comment]
Modern zabbix version can do all this json stuff itself. Why install python and additional scripts in papercut server?

I added these two api endpoints and rewrote the discovery with dependent items and JSONpath.

- `http://{HOST.DNS}:{$PAPERCUT_PORT}/api/health/devices`
- `http://{HOST.DNS}:{$PAPERCUT_PORT}/api/health/printers`

License, uptime, etc. can be pulled by `http://{HOST.DNS}:{$PAPERCUT_PORT}/api/health` endpoint with use of dependent items, too.

No additional software (except zabbix agent for server metrics) on papercut server and same values as this template.

**Comment by Trikke76** [comment]
a new version was published several months ago 
https://github.com/OpensourceICTSolutions/ZabbixPapercutNG

papercut should have updated the documentation they had it tested etc ...

**Comment by Kimotu** [comment]
@Trikke76 Right. They should have updated the documentation. I tried some hours to get this python crap working which failed with HTTP/403 each time, while authorization was right and I even added authorization headers and I spent some time to rework this template for json discovery.
But I am a bit proud, that the new template uses quite the same way of discovery like mine.

**Comment by Trikke76** [comment]
yes many ppl have changed at papercut and last time I had to do everything with someone from the helpdesk
i also dont work at this company anymore so i cant really update the git repo or close this ticket 
maybe it will help other users with similar issues
